### PR TITLE
[improve][txn] PIP-160 admin api coordinatorStats append batch info

### PR DESCRIPTION
**Pull request by poorbarcode**
Master Issue: #15370

### Motivation

see #15370

### Modifications

I will complete proposal #15370 with these pull requests( *current pull request is a part of step-6* ): 

1. Write the batch transaction log handler: `TxnLogBufferedWriter`
2. Configuration changes and protocol changes.
3. Transaction log store enables the batch feature.
4. Pending ack log store enables the batch feature.
5. Supports dynamic configuration.
6. Append admin API for transaction batch log and docs( admin and configuration doc ).
   GET /admin/v3/transactions/coordinatorStats
   GET /admin/v3/transactions/pendingAckStats/:tenant/:namespace:/:topic:/:subName
7. Append metrics support for transaction batch log.

### Documentation

- [ ] `doc-required` 
  
- [x] `doc-not-needed` 
  
- [ ] `doc` 

- [ ] `doc-complete`

### CODE REVIEW

**Review by codelipenghui — APPROVED** [review]
LGTM, please check the failed test.

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TxnBufferedWriterStat.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    /** If enabled the feature-batch, this attribute means maximum log records count in a batch. **/
+    private int batchedWriteMaxRecords = 512;
+
+    /** If enabled the feature-batch, this attribute means bytes size in a batch. **/
+    private int batchedWriteMaxSize;
+
+    /**
+     * If enabled the feature-batch, this attribute means maximum wait time(in millis) for the first record in a batch.
+     */
+    private int batchedWriteMaxDelayInMillis;
```
Comment (Technoboy-) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TxnBufferedWriterStat.java:38]: Please help check the naming of the above fields @momo-jun

**Review by liangyepianzhou — COMMENTED** [review]
I do not understand why the users need the admin API to get the configuration of the transaction batch. The configuration is what the user configures in broker.conf, isn't it?

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TxnBufferedWriterStat.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import lombok.Data;
+
+@Data
+public class TxnBufferedWriterStat {
+
+    /** Whether to enable the batch feature when writing to Bookie. **/
+    private boolean batchEnabled = false;
+
+    /** If enabled the feature-batch, this attribute means maximum log records count in a batch. **/
```
Comment (momo-jun) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TxnBufferedWriterStat.java:29]:
```suggestion
    /** If the batch feature is enabled, this attribute means the maximum count of log records in a batch. **/
```

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TxnBufferedWriterStat.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    private int batchedWriteMaxRecords = 512;
+
+    /** If enabled the feature-batch, this attribute means bytes size in a batch. **/
+    private int batchedWriteMaxSize;
+
+    /**
+     * If enabled the feature-batch, this attribute means maximum wait time(in millis) for the first record in a batch.
```
Comment (momo-jun) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TxnBufferedWriterStat.java:36]:
```suggestion
     * If the batch feature is enabled, this attribute means the maximum wait time (in milliseconds) for the first record in a batch.
```

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TxnBufferedWriterStat.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+@Data
+public class TxnBufferedWriterStat {
+
+    /** Whether to enable the batch feature when writing to Bookie. **/
+    private boolean batchEnabled = false;
+
+    /** If enabled the feature-batch, this attribute means maximum log records count in a batch. **/
+    private int batchedWriteMaxRecords = 512;
+
+    /** If enabled the feature-batch, this attribute means bytes size in a batch. **/
```
Comment (momo-jun) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TxnBufferedWriterStat.java:32]:
```suggestion
    /** If the batch feature is enabled, this attribute means the maximum size of bytes in a batch. **/
```

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TxnBufferedWriterStat.java (hunk 1 above)
Comment (momo-jun) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/TxnBufferedWriterStat.java:38]: I made a few suggestions for the annotation.

**Comment by poorbarcode** [comment]
> I do not understand why the users need the admin API to get the configuration of the transaction batch. The configuration is what the user configures in broker.conf, isn't it?

OK, I have make this PR to "draft"

**Pull request closed without merging**